amethyst_animation/src/systems/sampling.rs

```diff
@@ -76,28 +75,34 @@ where
                     process_sampler(control, sampler, &time, &mut self.inner);
                 }
             }
-            if !self.inner.is_empty() {
-                self.channels.clear();
-                self.channels
-                    .extend(self.inner.iter().map(|o| &o.1).unique().cloned());
-                for channel in &self.channels {
-                    match comp.blend_method(channel) {
-                        None => {
-                            if let Some(p) = self
-                                .inner
-                                .iter()
-                                .filter(|p| p.1 == *channel)
-                                .map(|p| p.2.clone())
-                                .last()
-                            {
-                                comp.apply_sample(channel, &p, &apply_data);
-                            }
+
```

Comment:
I would really separate the dependency changes from the CI changes. If you could close this PR and make two separate ones that would be great!